### PR TITLE
fix(stations): key Send-Groups-to-Stations assignments by roster id, not name

### PR DESCRIPTION
**Pull request by OPS-PIvers**
## Root cause

`buildStationsFromRandomGroups()` in `components/widgets/Stations/nexus.ts` always keyed the returned `assignments` map by raw student **display name**. The Randomizer's "Send Groups → Stations" action (`components/widgets/random/RandomSettings.tsx`) writes that map straight into the target Stations widget's config via `updateWidget()`, bypassing the Stations widget's own `coalesceLegacyKeys()` id-migration step (which only runs on the Stations widget's *own* persist path — drag/rotate/shuffle/reset — not on config written by another widget). Since `assignments` isn't PII-scrubbed for class-roster mode, this name-keyed map reached Firestore unscrubbed every time a teacher used "Send Groups → Stations" — including for ordinary **class-roster** dashboards, not just custom-roster mode. Confirmed live: `RandomSettings.tsx` is wired to a real "Send" button.

## Fix

`buildStationsFromRandomGroups()` now accepts an optional `rosterNameToId: Map<string, string>`; when a name resolves, the assignment is keyed by the roster student id instead of the raw name (falling back to name only when unresolved — custom-roster mode, same pre-existing limitation as before). `RandomSettings.tsx` builds this map by mirroring the Stations widget's own active-roster resolution logic (`rosters.find(r => r.id === activeRosterId) ?? rosters[0]`, `${firstName} ${lastName}`.trim()) before calling `buildStationsFromRandomGroups`.

## Band-aid rejected

A full stable-id scheme for custom-roster mode (a parallel `customRosterIds` array synced across every roster-edit UI action in both Stations and LunchCount) — investigated fully but genuinely architectural, spans 2 widgets, and risks new assignment-mismatch bugs if rushed. Left open as a backlog item; this PR fixes the more severe class-roster-mode leak via the cross-widget write path.

## Regression test

`tests/components/widgets/Stations/nexus.test.ts` — new case asserting assignments are keyed by roster id when a name→id map is provided, and that the raw name never appears in the output.

## Evidence

Fail-before (pre-fix `nexus.ts` via `git show dev-paul:...`): `AssertionError: expected undefined to be defined` (1 failed, 11 passed). Pass-after: 12/12.

Independently re-verified by the orchestrator: full `pnpm run validate` (626 root test files/7463 tests, 41 functions files/822 tests, all green) + `pnpm run build` — both clean on a second, independent run.

## Risk

**Contained** — two-file diff (`Stations/nexus.ts` + `random/RandomSettings.tsx`) plus a test, backward-compatible fallback to name-keying preserved for custom-roster mode.

## Related backlog

Companion PR (State & Data, same run) closes a related-but-distinct PII gap in `utils/dashboardPII.ts`'s scrubbing of custom-mode `assignments`. No file overlap between the two.

---
_Generated by [Claude Code](https://claude.ai/code/session_01JXVWwoBL8m4Wh7PkrrXVMM)_

### CODE REVIEW

**Comment by OPS-PIvers** [comment]
Good catch — fixed in `a0ed2012`. `rosterNameToId` construction now also requires the *source* Random widget's own `rosterMode !== 'custom'`, not just the target Stations widget's. Extracted the combined gate into a small, independently-testable pure function (`shouldResolveRosterNames(sourceRosterMode, targetRosterMode)` in `nexus.ts`) with 5 new unit tests covering all 4 mode combinations plus the `undefined`-defaults case. Full `pnpm run validate` + `pnpm run build` re-verified green (626 root test files/7468 tests, 41 functions files/822 tests).

---
_Generated by [Claude Code](https://claude.ai/code)_

File: components/widgets/Stations/nexus.ts [hunk 1]
```diff
@@ -11,17 +11,35 @@ const UUID_RE =
...
+ *                        is keyed by the roster id instead of the raw name —
+ *                        matching the id-keyed scheme the Stations widget
+ *                        itself writes via `coalesceLegacyKeys`. Without this,
+ *                        assignments are keyed by raw student display name,
+ *                        which reaches Firestore unscrubbed (PII_WIDGET_FIELDS
+ *                        has no way to strip a value-keyed map). Names that
+ *                        don't resolve (custom-roster mode, or no map passed)
+ *                        fall back to the legacy name-keyed entry.
  */
+/** Only trust a Random-group name→roster-id resolution when BOTH the source (Random) and target (Stations) widgets are in class-roster mode — else a freeform-typed Random name could collide with an unrelated real student's name. */
```
[the remark this answers is not in the document]
Comment (OPS-PIvers) [reply components/widgets/Stations/nexus.ts]: No code change needed — this was already fixed on the branch in `24ec261` ("fix(random): move buildStationsFromRandomGroups's JSDoc back above itself"), which landed after this comment was written. Verified on the current head (`24ec261`): `shouldResolveRosterNames` now carries its own one-line comment and the `@param groups` / `@param sharedGroups` / `@param rosterNameToId` block sits directly above `buildStationsFromRandomGroups`.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by OPS-PIvers — COMMENTED** [review]
## Automated Code Review

**Date:** 2026-08-19

### Summary
Keys `Send Groups → Stations` assignments by roster student id instead of raw display name, gated behind a new `shouldResolveRosterNames(source, target)` predicate that requires *both* the Random widget and the target Stations widget to be in class-roster mode. Verified locally on head `24ec261`: `pnpm type-check` exit 0, targeted `eslint --max-warnings 0` clean, `tests/components/widgets/Stations/nexus.test.ts` 17/17 passing. The core change is correct and well-tested; one edge case in the roster-lookup fallback is worth tightening.

### Code Quality
✓ No issues found. `shouldResolveRosterNames` is a pure, exported, independently-testable predicate — good factoring, and the `react-refresh/only-export-components` rationale for keeping it in `nexus.ts` still holds.

One note on the `buildStationsFromRandomGroups` JSDoc: this diff grows it from 8 lines to 18. CLAUDE.md's comment rule is "one short line max — never multi-paragraph docstrings," and this PR's own follow-up commit `e45682d`-style sweeps on sibling nightly branches are trimming exactly this shape. The block was pre-existing so this isn't a regression, but the `@param rosterNameToId` addition is 10 new lines of the thing the convention forbids. The load-bearing fact is one line: *"roster id keys are PII-safe and match `coalesceLegacyKeys`; unresolved names fall back to name keys."*

### Widget Standards
✓ Not applicable — no new widget, no registry/appearance/scaling surface touched.

### Admin Config Standards
✓ Not applicable.

### Firestore & Security
✓ No issues found, and this is the PR's actual point. Keying by roster id is what lets `PII_WIDGET_FIELDS` scrubbing work at all — a value-keyed map of student display names has no field-level scrub path, so this genuinely removes PII from the Firestore write for the class-roster case. Complements PR #2496, which handles the remaining custom-mode case by scrubbing `assignments` outright.

### Regression Risk
Low overall, with one real edge case:

**`rosters[0]` fallback can silently key against the wrong class** — `RandomSettings.tsx:113-114`:
```ts
const targetRoster = rosters.find((r) => r.id === activeRosterId) ?? rosters[0];
```
The `?? rosters[0]` fallback is inconsistent with how this same file resolves the roster everywhere else — `activeRoster` at `RandomSettings.tsx:141-143` is `rosters.find((r) => r.id === activeRosterId)` with **no** fallback, and `importFromRoster` early-returns when it's undefined. So when `activeRosterId` is stale or unset and the teacher has more than one roster, `handleSendToStations` builds its name→id map from an arbitrary *other* class. A picked name that happens to match a student in that unrelated roster gets keyed to that student's id — which is the same misattribution failure mode `shouldResolveRosterNames` was added in `a0ed201` to prevent, just reached by a different route.

Suggest dropping the fallback so it matches the sibling lookup and degrades to name keys (the safe legacy path) rather than to a guess:
```ts
const targetRoster = rosters.find((r) => r.id === activeRosterId);
```

Other regression surfaces checked and clear:
- `StationsConfig` has no per-widget `rosterId`; both widgets read the same dashboard-level `activeRosterId`, so the source/target roster is genuinely the same one — the predicate's premise holds.
- `rosterMode` defaults to `'class'` in `config/widgetDefaults.ts` for all four roster widgets, so `shouldResolveRosterNames`' `!== 'custom'` treatment of `undefined` is correct rather than accidentally permissive.
- `buildStationsFromRandomGroups`' third parameter is optional, so existing call sites are unaffected.
- No `WidgetRegistry.ts`, `DashboardContext.tsx`, `types.ts`, or `functions/` changes.

### Missing Tests
Adequate for what's implemented. `nexus.test.ts` covers the predicate's four mode combinations and the id-vs-name keying both ways. The gap is the `rosters[0]` fallback above — it lives in `RandomSettings.tsx`, which has no test, so a stale-`activeRosterId` case isn't covered anywhere. If you take the one-line fix, a small test asserting name-key fallback when `activeRosterId` matches nothing would pin it.

### Overall
**Merge readiness:** Ready with minor notes
Correct, focused, well-tested PII fix; the `?? rosters[0]` fallback is a one-line change that closes the last route to the exact misattribution this PR set out to prevent.

---
_This review was generated automatically. All findings should be verified by a human before merging._

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by OPS-PIvers** [comment]
Good catch — fixed in `46c50b21`. Dropped the `?? rosters[0]` fallback so it matches how `activeRoster` is resolved everywhere else in this file (no fallback, degrades to the safe name-key path when unresolved). Re-verified: full `pnpm run test` (626 files/7468 tests, unchanged from before) + `pnpm run build` both green. Skipped adding a dedicated `RandomSettings.tsx` component test for this specific case — proportionate to a one-line fix with no existing test harness for this file, per the "proportionate verification for a proportionately small change" precedent already established in this repo's nightly routine.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by OPS-PIvers** [comment]
Good catch on the stale comment — `Widget.tsx` isn't otherwise touched by this PR, so logged it as a backlog item in `docs/routines/debugger.md` rather than expanding this diff's scope, per your own note ("worth a follow-up tweak whenever that file is next touched, not blocking").

---
_Generated by [Claude Code](https://claude.ai/code)_
